AnsiFeed/AnsiTextColored.cpp

```diff
@@ -0,0 +1,562 @@
+
+// Based on https://gist.github.com/ddovod/be210315f285becc6b0e455b775286e1
+
+#include <vector>
+#include <string>
+#define IMGUI_DEFINE_MATH_OPERATORS
+#include <imgui_internal.h>
+
+namespace jet
+{
+    std::vector<std::string> split(const std::string& str, const std::string& delim = " ")
+    {
+        std::vector<std::string> res;
+        std::size_t current, previous = 0;
+        current = str.find(delim);
+        while (current != std::string::npos) {
+            res.push_back(str.substr(previous, current - previous));
+            previous = current + delim.length();
+            current = str.find(delim, previous);
+        }
+        res.push_back(str.substr(previous, current - previous));
+        return res;
+    }
+}
+
+namespace ImGui
+{
+    static const int kMaxChar = 10000;
+    static char char_buf[kMaxChar];
+    static ImU32 col_buf[kMaxChar];
+    static bool char_skip[kMaxChar];
+
+    bool ParseColor(const char* s, ImU32* col, int* skipChars)
+    {
+        // if (s[0] != '\033' && s[0] != '\x1b')// && s[0] != '\e')
+        // {
+        //     return false;
+        // }
+
+        // if (s[1] != '[') {
+        //     return false;
+        // }
+
+        if (s[0] != '\033' || s[1] != '[')
+        {
+            return false;
+        }
+
+        if (s[2] == 'm') {
+            *col = 0xffcccccc;
+            *skipChars = 3;
+            return true;
+        }
+
+        if (s[2] == '0' && s[3] == 'm') {
+            *col = 0xffcccccc;
+            *skipChars = 4;
+            return true;
+        }
+
+        const char* seqEnd = &s[2];
+        while (*seqEnd != 'm') {
+            seqEnd++;
+        }
+
+        std::string seq( &s[2], seqEnd );
+        std::string colorStr;
+        for (const auto& el : jet::split(seq, ";")) {
```

Comment:
CI build is failing here on the `const auto&`. This syntax is probably too new; it should be replaced with a less-modern substitute.